File: 2010/20100923_python_reversi/reversi.py
class Reversi(object):


    def __init__(self):
        self.tabuleiro = [
            [' ', ' ', ' ', ' '],
            [' ', 'P', 'B', ' '],
            [' ', 'B', 'P', ' '],
            [' ', ' ', ' ', ' '],
        ]

    def jogada(self, jogador, posicao):
        linha  = posicao[0]
        coluna = posicao[1]
        self.tabuleiro[linha][coluna] = jogador
        if coluna == 0:
            self.tabuleiro[linha][coluna + 1] = jogador
        elif coluna == 3:
            self.tabuleiro[linha][coluna - 1] = jogador

        if linha == 0:
            self.tabuleiro[linha + 1][coluna] = jogador
        elif linha == 3:
            self.tabuleiro[linha - 1][coluna] = jogador

File: 2010/20100720_java_reversi_substituicoes/src/Reversi.java


public class Reversi {

    public int[][] resultado(int[][] jogo, Jogada jogada){


        int[][] jogoResultado = jogo;
        jogoResultado[jogada.linha][jogada.coluna] = jogada.jogador;

        for (int i = jogada.linha-1; i>= 0;i--){
            if (jogoResultado[i][jogada.coluna] == jogada.jogador){
                break;
            }
            if (jogoResultado[i][jogada.coluna] == jogada.outroJogador()){
                jogoResultado[i][jogada.coluna] = jogada.jogador;
            }
            if (jogoResultado[i][jogada.coluna] == -1){
                for (int j = i+1; j<jogada.linha; j++){
                    jogoResultado[j][jogada.coluna] = jogada.outroJogador();
                }
                break;
            }
            


        }






        jogoResultado[jogada.linha][jogada.coluna] = jogada.jogador;
        if (jogada.jogador == 1)
            jogoResultado[jogada.linha - 1][jogada.coluna + 1] = jogada.jogador;
        else if (jogada.linha == 4)       
            jogoResultado[jogada.linha][jogada.coluna + 1] = jogada.jogador;

        return jogoResultado;
    }

}
